src/WorldSelector.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+
+import React from 'react';
+import type { WorldName } from './types';
+import type { IntlShape } from 'react-intl';
+import { injectIntl } from 'react-intl';
+
+import {ReactComponent as HeaderIcon} from './svg/SelectWorld.svg'
+import {ReactComponent as RabbitIcon} from './svg/Rabbit.svg';
+import {ReactComponent as RobotIcon} from './svg/Robot.svg';
+import { ReactComponent as SpaceShipIcon } from './svg/SpaceShip.svg';
+
+import './WorldSelector.scss';
+
+type WorldSelectorProps = {
+    intl: IntlShape,
+    onSelect: (value: WorldName) => void
+};
+
+type WorldSelectorState = {
+    world: WorldName
+}
+
+class WorldSelector extends React.Component<WorldSelectorProps, WorldSelectorState> {
+    constructor (props) {
+        super(props);
+        this.state = {
+            world: 'default'
+        }
+    }
+
+    componentDidUpdate(prevProps: WorldSelectorProps, prevState: WorldSelectorState) {
+        if (this.state.world !== prevState.world) {
+            this.props.onSelect(this.state.world);
+        }
+    }
+
+    handleCharacterClick = (event: Event, worldName: WorldName) => {
+        event.preventDefault();
+        this.setState({ world: worldName });
+    }
+
+    handleCharacterKeyDown = (event: KeyboardEvent, worldName: WorldName) => {
+        const toggleKeys = [' ', 'Enter'];
+
+        if (toggleKeys.indexOf(event.key) !== -1) {
+            event.preventDefault();
+            this.setState({ world: worldName });
+        }
+    }
+
+    render() {
+        return (
+            <div className='WorldSelector'>
+                <HeaderIcon
+                    aria-label={this.props.intl.formatMessage({id:'WorldSelector.world'})}
+                    className="HeaderIcon"
+                />
+                <RobotIcon
+                    aria-label={this.props.intl.formatMessage({id:'WorldSelector.world.default'})}
+                    className={"WorldIcon" + (this.state.world === 'default' ? " WorldIcon--selected" : "") }
+                    onClick={(e: Event) => { this.handleCharacterClick(e, 'default')}}
```

Comment:
Using arrow function inside of render method is not recommended, because this will create a new function each time the component renders. We can simplify this by assigning value property to each Character buttons, and get this value from handleCharacterClick. 

`onClick={this.handleCharacterClick}`

Reply:
Fixed.